docs/src/design.md

```diff
@@ -72,9 +72,14 @@ Here are the current [`Model`](@ref) traits in use and their possible values:
  - [`output_type`](@ref) - determines how many output variates a [`Model`](@ref) can learn
    - [`SingleOutput`](@ref): Fits and predicts on a single output only.
    - [`MultiOutput`](@ref): Fits and predicts on multiple outputs at a time.
+ - [`predict_input_type`](@ref) - determines which datatypes a [`Model`](@ref) can accept at predict time.
+   - [`PointPredictInput`](@ref): Real valued input variables accepted at predict time.
+   - [`DistributionPredictInput`](@ref): Distributions over input variables accepted at predict time.
```

Comment:
On reflection I'm not sure we need `DistributionPredictInput` (only `PointOrDistributionPredictInput`). I can't think of a use case where we would be able to fit a model on point inputs, but only be able to predict it on distribution inputs. 

Removing this would also simplify certain things in the test utils, because it would mean that this PR supports only strict additions to the current API functionality (i.e. ability to call `predict` with a distribution input), and not a partial addition with a partial removal (i.e.  ability to call `predict` with a distribution input, but not with point input, which was previously guaranteed).

Reply:
Settled on removing the `DistributionPredictInput` case, firstly because it does indeed simplify things in the test utils.  And secondly I think the logic of avoiding a partial addition and partial removal is sound.  We are trying to make a non-breaking change after all.